### ISSUE TITLE
oqmd dataset generator error 

### ISSUE THREAD

**Issue by WdBlink** [Tony-Y/cgnn#1]
Hello developer, thanks for your code and research. It's helped a lot. But there is a question when I run `python ../tools/oqmd_data.py`, could you teach me how to solve this problem？ 
> Traceback (most recent call last):
  File "../tools/oqmd_data.py", line 128, in <module>
    main(**options)
  File "../tools/oqmd_data.py", line 110, in main
    entries = get_valid_entries()
  File "../tools/oqmd_data.py", line 34, in get_valid_entries
    return list(entries)
  File "/home/dell/anaconda3/envs/ElemNet/lib/python2.7/site-packages/django/db/models/query.py", line 162, in __iter__
    self._fetch_all()
  File "/home/dell/anaconda3/envs/ElemNet/lib/python2.7/site-packages/django/db/models/query.py", line 965, in _fetch_all
    self._result_cache = list(self.iterator())
  File "/home/dell/anaconda3/envs/ElemNet/lib/python2.7/site-packages/django/db/models/query.py", line 1217, in iterator
    for row in compiler.results_iter():
  File "/home/dell/anaconda3/envs/ElemNet/lib/python2.7/site-packages/django/db/models/sql/compiler.py", line 794, in results_iter
    results = self.execute_sql(MULTI)
  File "/home/dell/anaconda3/envs/ElemNet/lib/python2.7/site-packages/django/db/models/sql/compiler.py", line 840, in execute_sql
    cursor.execute(sql, params)
  File "/home/dell/anaconda3/envs/ElemNet/lib/python2.7/site-packages/django/db/backends/utils.py", line 64, in execute
    return self.cursor.execute(sql, params)
  File "/home/dell/anaconda3/envs/ElemNet/lib/python2.7/site-packages/django/db/utils.py", line 98, in __exit__
    six.reraise(dj_exc_type, dj_exc_value, traceback)
  File "/home/dell/anaconda3/envs/ElemNet/lib/python2.7/site-packages/django/db/backends/utils.py", line 64, in execute
    return self.cursor.execute(sql, params)
  File "/home/dell/anaconda3/envs/ElemNet/lib/python2.7/site-packages/django/db/backends/mysql/base.py", line 124, in execute
    return self.cursor.execute(query, args)
  File "/home/dell/anaconda3/envs/ElemNet/lib/python2.7/site-packages/MySQLdb/cursors.py", line 205, in execute
    self.errorhandler(self, exc, value)
  File "/home/dell/anaconda3/envs/ElemNet/lib/python2.7/site-packages/MySQLdb/connections.py", line 36, in defaulterrorhandler
    raise errorclass, errorvalue
django.db.utils.ProgrammingError: (1146, "Table 'oqmd_v1_2.calculations' doesn't exist")

**Comment by Tony-Y** [comment]
> django.db.utils.ProgrammingError: (1146, "Table 'oqmd_v1_2.calculations' doesn't exist")

You seem to have a problem of the OQMD database. You can check whether the database exists as the following:

```
mysql> SHOW DATABASES LIKE 'oqmd%';
+------------------+
| Database (oqmd%) |
+------------------+
| oqmd_v1_2        |
+------------------+
```

**Comment by WdBlink** [comment]
Thanks for your reply. And the database exists:
```
mysql> SHOW DATABASES LIKE 'oqmd%';
+------------------+
| Database (oqmd%) |
+------------------+
| oqmd_v1_2        |
+------------------+
1 row in set (0.00 sec)
```

**Comment by Tony-Y** [comment]
You can check whether the table `oqmd_v1_2.calculations` exists as the following:
```
mysql> SHOW TABLES FROM oqmd_v1_2 LIKE 'calculations';
+------------------------------------+
| Tables_in_oqmd_v1_2 (calculations) |
+------------------------------------+
| calculations                       |
+------------------------------------+
```

**Comment by WdBlink** [comment]
I found that when I used `zcat ./qmdb__v1_2__062018.sql.gz | mysql -u $user -p $password oqmd_v1_2` . I can't restore the data and got this:
```
>> zcat ./qmdb__v1_2__062018.sql.gz | mysql -u dell -p dell_123456 oqmd_v1_2

mysql  Ver 14.14 Distrib 5.7.29, for Linux (x86_64) using  EditLine wrapper
Copyright (c) 2000, 2020, Oracle and/or its affiliates. All rights reserved.

Oracle is a registered trademark of Oracle Corporation and/or its
affiliates. Other names may be trademarks of their respective
owners.

Usage: mysql [OPTIONS] [database]
  -?, --help          Display this help and exit.
  -I, --help          Synonym for -?
  --auto-rehash       Enable automatic rehashing. One doesn't need to use
                      'rehash' to get table and field completion, but startup
                      and reconnecting may take a longer time. Disable with
                      --disable-auto-rehash.
                      (Defaults to on; use --skip-auto-rehash to disable.)
  -A, --no-auto-rehash
                      No automatic rehashing. One has to use 'rehash' to get
                      table and field completion. This gives a quicker start of
                      mysql and disables rehashing on reconnect.
  --auto-vertical-output
                      Automatically switch to vertical output mode if the
                      result is wider than the terminal width.
  -B, --batch         Don't use history file. Disable interactive behavior.
                      (Enables --silent.)
  --bind-address=name IP address to bind to.
  --binary-as-hex     Print binary data as hex
  --character-sets-dir=name
                      Directory for character set files.
  --column-type-info  Display column type information.
  -c, --comments      Preserve comments. Send comments to the server. The
                      default is --skip-comments (discard comments), enable
                      with --comments.
  -C, --compress      Use compression in server/client protocol.
  -#, --debug[=#]     This is a non-debug version. Catch this and exit.
  --debug-check       This is a non-debug version. Catch this and exit.
  -T, --debug-info    This is a non-debug version. Catch this and exit.
  -D, --database=name Database to use.
  --default-character-set=name
                      Set the default character set.
  --delimiter=name    Delimiter to be used.
  --enable-cleartext-plugin
                      Enable/disable the clear text authentication plugin.
  -e, --execute=name  Execute command and quit. (Disables --force and history
                      file.)
  -E, --vertical      Print the output of a query (rows) vertically.
  -f, --force         Continue even if we get an SQL error.
  --histignore=name   A colon-separated list of patterns to keep statements
                      from getting logged into syslog and mysql history.
  -G, --named-commands
                      Enable named commands. Named commands mean this program's
                      internal commands; see mysql> help . When enabled, the
                      named commands can be used from any line of the query,
                      otherwise only from the first line, before an enter.
                      Disable with --disable-named-commands. This option is
                      disabled by default.
  -i, --ignore-spaces Ignore space after function names.
  --init-command=name SQL Command to execute when connecting to MySQL server.
                      Will automatically be re-executed when reconnecting.
  --local-infile      Enable/disable LOAD DATA LOCAL INFILE.
  -b, --no-beep       Turn off beep on error.
  -h, --host=name     Connect to host.
  -H, --html          Produce HTML output.
  -X, --xml           Produce XML output.
  --line-numbers      Write line numbers for errors.
                      (Defaults to on; use --skip-line-numbers to disable.)
  -L, --skip-line-numbers
                      Don't write line number for errors.
  -n, --unbuffered    Flush buffer after each query.
  --column-names      Write column names in results.
                      (Defaults to on; use --skip-column-names to disable.)
  -N, --skip-column-names
                      Don't write column names in results.
  --sigint-ignore     Ignore SIGINT (CTRL-C).
  -o, --one-database  Ignore statements except those that occur while the
                      default database is the one named at the command line.
  --pager[=name]      Pager to use to display results. If you don't supply an
                      option, the default pager is taken from your ENV variable
                      PAGER. Valid pagers are less, more, cat [> filename],
                      etc. See interactive help (\h) also. This option does not
                      work in batch mode. Disable with --disable-pager. This
                      option is disabled by default.
  -p, --password[=name]
                      Password to use when connecting to server. If password is
                      not given it's asked from the tty.
  -P, --port=#        Port number to use for connection or 0 for default to, in
                      order of preference, my.cnf, $MYSQL_TCP_PORT,
                      /etc/services, built-in default (3306).
  --prompt=name       Set the mysql prompt to this value.
  --protocol=name     The protocol to use for connection (tcp, socket, pipe,
                      memory).
  -q, --quick         Don't cache result, print it row by row. This may slow
                      down the server if the output is suspended. Doesn't use
                      history file.
  -r, --raw           Write fields without conversion. Used with --batch.
  --reconnect         Reconnect if the connection is lost. Disable with
                      --disable-reconnect. This option is enabled by default.
                      (Defaults to on; use --skip-reconnect to disable.)
  -s, --silent        Be more silent. Print results with a tab as separator,
                      each row on new line.
  -S, --socket=name   The socket file to use for connection.
  --ssl-mode=name     SSL connection mode.
  --ssl               Deprecated. Use --ssl-mode instead.
                      (Defaults to on; use --skip-ssl to disable.)
  --ssl-verify-server-cert
                      Deprecated. Use --ssl-mode=VERIFY_IDENTITY instead.
  --ssl-ca=name       CA file in PEM format.
  --ssl-capath=name   CA directory.
  --ssl-cert=name     X509 cert in PEM format.
  --ssl-cipher=name   SSL cipher to use.
  --ssl-key=name      X509 key in PEM format.
  --ssl-crl=name      Certificate revocation list.
  --ssl-crlpath=name  Certificate revocation list path.
  --tls-version=name  TLS version to use, permitted values are: TLSv1, TLSv1.1,
                      TLSv1.2
  --server-public-key-path=name
                      File path to the server public RSA key in PEM format.
  --get-server-public-key
                      Get server public key
  -t, --table         Output in table format.
  --tee=name          Append everything into outfile. See interactive help (\h)
                      also. Does not work in batch mode. Disable with
                      --disable-tee. This option is disabled by default.
  -u, --user=name     User for login if not current user.
  -U, --safe-updates  Only allow UPDATE and DELETE that uses keys.
  -U, --i-am-a-dummy  Synonym for option --safe-updates, -U.
  -v, --verbose       Write more. (-v -v -v gives the table output format).
  -V, --version       Output version information and exit.
  -w, --wait          Wait and retry if connection is down.
  --connect-timeout=# Number of seconds before connection timeout.
  --max-allowed-packet=#
                      The maximum packet length to send to or receive from
                      server.
  --net-buffer-length=#
                      The buffer size for TCP/IP and socket communication.
  --select-limit=#    Automatic limit for SELECT when using --safe-updates.
  --max-join-size=#   Automatic limit for rows in a join when using
                      --safe-updates.
  --secure-auth       Refuse client connecting to server if it uses old
                      (pre-4.1.1) protocol. Deprecated. Always TRUE
  --server-arg=name   Send embedded server this as a parameter.
  --show-warnings     Show warnings after every statement.
  -j, --syslog        Log filtered interactive commands to syslog. Filtering of
                      commands depends on the patterns supplied via histignore
                      option besides the default patterns.
  --plugin-dir=name   Directory for client-side plugins.
  --default-auth=name Default authentication client-side plugin to use.
  --binary-mode       By default, ASCII '\0' is disallowed and '\r\n' is
                      translated to '\n'. This switch turns off both features,
                      and also turns off parsing of all clientcommands except
                      \C and DELIMITER, in non-interactive mode (for input
                      piped to mysql or loaded using the 'source' command).
                      This is necessary when processing output from mysqlbinlog
                      that may contain blobs.
  --connect-expired-password
                      Notify the server that this client is prepared to handle
                      expired password sandbox mode.

Default options are read from the following files in the given order:
/etc/my.cnf /etc/mysql/my.cnf ~/.my.cnf
The following groups are read: mysql client
The following options may be given as the first argument:
--print-defaults        Print the program argument list and exit.
--no-defaults           Don't read default options from any option file,
                        except for login file.
--defaults-file=#       Only read default options from the given file #.
--defaults-extra-file=# Read this file after the global files are read.
--defaults-group-suffix=#
                        Also read groups with concat(group, suffix)
--login-path=#          Read this path from the login file.

Variables (--variable-name=value)
and boolean options {FALSE|TRUE}  Value (after reading options)
--------------------------------- ----------------------------------------
auto-rehash                       TRUE
auto-vertical-output              FALSE
bind-address                      (No default value)
binary-as-hex                     FALSE
character-sets-dir                (No default value)
column-type-info                  FALSE
comments                          FALSE
compress                          FALSE
database                          (No default value)
default-character-set             auto
delimiter                         ;
enable-cleartext-plugin           FALSE
vertical                          FALSE
force                             FALSE
histignore                        (No default value)
named-commands                    FALSE
ignore-spaces                     FALSE
init-command                      (No default value)
local-infile                      FALSE
no-beep                           FALSE
host                              (No default value)
html                              FALSE
xml                               FALSE
line-numbers                      TRUE
unbuffered                        FALSE
column-names                      TRUE
sigint-ignore                     FALSE
port                              0
prompt                            mysql>
quick                             FALSE
raw                               FALSE
reconnect                         FALSE
socket                            (No default value)
ssl                               TRUE
ssl-verify-server-cert            FALSE
ssl-ca                            (No default value)
ssl-capath                        (No default value)
ssl-cert                          (No default value)
ssl-cipher                        (No default value)
ssl-key                           (No default value)
ssl-crl                           (No default value)
ssl-crlpath                       (No default value)
tls-version                       (No default value)
server-public-key-path            (No default value)
get-server-public-key             FALSE
table                             FALSE
user                              dell
safe-updates                      FALSE
i-am-a-dummy                      FALSE
connect-timeout                   0
max-allowed-packet                16777216
net-buffer-length                 16384
select-limit                      1000
max-join-size                     1000000
secure-auth                       TRUE
show-warnings                     FALSE
plugin-dir                        (No default value)
default-auth                      (No default value)
binary-mode                       FALSE
connect-expired-password          FALSE
```
So I used the other way to restore the data which is `mysql>Source /home/dell/data/qmdb__v1_2__062018.sql.gz` The database seems to restored successfully but the tables still incomplete:
```
 python ../tools/oqmd_data.py
Traceback (most recent call last):
  File "../tools/oqmd_data.py", line 128, in <module>
    main(**options)
  File "../tools/oqmd_data.py", line 110, in main
    entries = get_valid_entries()
  File "../tools/oqmd_data.py", line 34, in get_valid_entries
    return list(entries)
  File "/home/dell/anaconda3/envs/ElemNet/lib/python2.7/site-packages/django/db/models/query.py", line 162, in __iter__
    self._fetch_all()
  File "/home/dell/anaconda3/envs/ElemNet/lib/python2.7/site-packages/django/db/models/query.py", line 965, in _fetch_all
    self._result_cache = list(self.iterator())
  File "/home/dell/anaconda3/envs/ElemNet/lib/python2.7/site-packages/django/db/models/query.py", line 1217, in iterator
    for row in compiler.results_iter():
  File "/home/dell/anaconda3/envs/ElemNet/lib/python2.7/site-packages/django/db/models/sql/compiler.py", line 794, in results_iter
    results = self.execute_sql(MULTI)
  File "/home/dell/anaconda3/envs/ElemNet/lib/python2.7/site-packages/django/db/models/sql/compiler.py", line 840, in execute_sql
    cursor.execute(sql, params)
  File "/home/dell/anaconda3/envs/ElemNet/lib/python2.7/site-packages/django/db/backends/utils.py", line 64, in execute
    return self.cursor.execute(sql, params)
  File "/home/dell/anaconda3/envs/ElemNet/lib/python2.7/site-packages/django/db/utils.py", line 98, in __exit__
    six.reraise(dj_exc_type, dj_exc_value, traceback)
  File "/home/dell/anaconda3/envs/ElemNet/lib/python2.7/site-packages/django/db/backends/utils.py", line 64, in execute
    return self.cursor.execute(sql, params)
  File "/home/dell/anaconda3/envs/ElemNet/lib/python2.7/site-packages/django/db/backends/mysql/base.py", line 124, in execute
    return self.cursor.execute(query, args)
  File "/home/dell/anaconda3/envs/ElemNet/lib/python2.7/site-packages/MySQLdb/cursors.py", line 205, in execute
    self.errorhandler(self, exc, value)
  File "/home/dell/anaconda3/envs/ElemNet/lib/python2.7/site-packages/MySQLdb/connections.py", line 36, in defaulterrorhandler
    raise errorclass, errorvalue
django.db.utils.ProgrammingError: (1146, "Table 'oqmd_v1_2.entries' doesn't exist")
```

```
mysql> use oqmd_v1_2
Reading table information for completion of table and column names
You can turn off this feature to get a quicker startup with -A

Database changed
mysql> show tables;
+----------------------------------+
| Tables_in_oqmd_v1_2              |
+----------------------------------+
| atoms                            |
| authors                          |
| calculations                     |
| compositions                     |
| django_content_type              |
| django_session                   |
| django_site                      |
| dos                              |
| elements                         |
| expt_formation_energies          |
| fits                             |
| fits_dft                         |
| fits_elements                    |
| fits_experiments                 |
| google_analytics_analytics       |
| google_analytics_googleanalytics |
| hosts                            |
| hubbard_corrections              |
| hubbards                         |
| journals                         |
| meta_data                        |
| projects                         |
| publications                     |
| publications_author_set          |
| reference_energies               |
| rotations                        |
| spacegroups                      |
| species                          |
| translations                     |
| users                            |
| vasp_potentials                  |
| wyckoffsites                     |
+----------------------------------+
32 rows in set (0.00 sec)
```

**Comment by Tony-Y** [comment]
Please check the file size. The OQMD official site says:
> qmdb__v1_2__062018.sql.gz (size: 8.9G)

Also, I found the following note:
> Important note: the new qmpy version (v1.3) does not support any of the older qmdb versions (qmdb < 1.3) out of the box.

So, you must install the appropriate version of `qmpy` as the following:
```
pip install qmpy==1.2.0
```

**Comment by WdBlink** [comment]
```
➜ ls -lh
-rw-rw-r--  1 dell dell  56G Jun 25  2018 qmdb__v1_2__062018.sql
-rw-rw-r--  1 dell dell 8.9G Jun 25  2018 qmdb__v1_2__062018.sql.gz
```
I think the raw data is fine. And after I changed the qmpy version. The question is still there T _ T:
```
➜ python ../tools/oqmd_data.py
/home/dell/anaconda3/envs/ElemNet/lib/python2.7/site-packages/qmpy/data/__init__.py:7: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.
  element_groups = yaml.load(data)

/home/dell/anaconda3/envs/ElemNet/lib/python2.7/site-packages/qmpy/data/__init__.py:10: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.
  elements = yaml.load(data)

/home/dell/anaconda3/envs/ElemNet/lib/python2.7/site-packages/qmpy/data/__init__.py:13: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.
  chem_pots = yaml.load(data)

/home/dell/anaconda3/envs/ElemNet/lib/python2.7/site-packages/qmpy/analysis/miedema.py:17: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.
  params = yaml.load(open(qmpy.INSTALL_PATH+'/data/miedema.yml').read())

/home/dell/anaconda3/envs/ElemNet/lib/python2.7/site-packages/qmpy/configuration/vasp_settings/inputs/__init__.py:9: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.
  settings =  yaml.load(open('%s/%s' % (vs_path, f)).read())

/home/dell/anaconda3/envs/ElemNet/lib/python2.7/site-packages/qmpy/configuration/vasp_settings/__init__.py:8: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.
  thubbards = yaml.load(open(vs_path+'/hubbards.yml').read())

/home/dell/anaconda3/envs/ElemNet/lib/python2.7/site-packages/qmpy/configuration/vasp_settings/__init__.py:20: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.
  POTENTIALS = yaml.load(open(vs_path+'/potentials.yml').read())

/home/dell/anaconda3/envs/ElemNet/lib/python2.7/site-packages/pymatgen/core/physical_constants.py:53: DeprecationWarning: The pymatgen.core.physical_constants module is deprecated and will be removed in pymatgen 4.0. Pls use scipy.constants.
  DeprecationWarning)

Traceback (most recent call last):
  File "../tools/oqmd_data.py", line 128, in <module>
    main(**options)
  File "../tools/oqmd_data.py", line 110, in main
    entries = get_valid_entries()
  File "../tools/oqmd_data.py", line 34, in get_valid_entries
    return list(entries)
  File "/home/dell/anaconda3/envs/ElemNet/lib/python2.7/site-packages/django/db/models/query.py", line 96, in __iter__
    self._fetch_all()
  File "/home/dell/anaconda3/envs/ElemNet/lib/python2.7/site-packages/django/db/models/query.py", line 857, in _fetch_all
    self._result_cache = list(self.iterator())
  File "/home/dell/anaconda3/envs/ElemNet/lib/python2.7/site-packages/django/db/models/query.py", line 1068, in iterator
    for row in self.query.get_compiler(self.db).results_iter():
  File "/home/dell/anaconda3/envs/ElemNet/lib/python2.7/site-packages/django/db/models/sql/compiler.py", line 713, in results_iter
    for rows in self.execute_sql(MULTI):
  File "/home/dell/anaconda3/envs/ElemNet/lib/python2.7/site-packages/django/db/models/sql/compiler.py", line 786, in execute_sql
    cursor.execute(sql, params)
  File "/home/dell/anaconda3/envs/ElemNet/lib/python2.7/site-packages/django/db/backends/util.py", line 69, in execute
    return super(CursorDebugWrapper, self).execute(sql, params)
  File "/home/dell/anaconda3/envs/ElemNet/lib/python2.7/site-packages/django/db/backends/util.py", line 53, in execute
    return self.cursor.execute(sql, params)
  File "/home/dell/anaconda3/envs/ElemNet/lib/python2.7/site-packages/django/db/utils.py", line 99, in __exit__
    six.reraise(dj_exc_type, dj_exc_value, traceback)
  File "/home/dell/anaconda3/envs/ElemNet/lib/python2.7/site-packages/django/db/backends/util.py", line 53, in execute
    return self.cursor.execute(sql, params)
  File "/home/dell/anaconda3/envs/ElemNet/lib/python2.7/site-packages/django/db/backends/mysql/base.py", line 124, in execute
    return self.cursor.execute(query, args)
  File "/home/dell/anaconda3/envs/ElemNet/lib/python2.7/site-packages/MySQLdb/cursors.py", line 205, in execute
    self.errorhandler(self, exc, value)
  File "/home/dell/anaconda3/envs/ElemNet/lib/python2.7/site-packages/MySQLdb/connections.py", line 36, in defaulterrorhandler
    raise errorclass, errorvalue
django.db.utils.ProgrammingError: (1146, "Table 'oqmd_v1_2.entries' doesn't exist")
```

**Comment by Tony-Y** [comment]
(1) Check the OQMD dataset in `mysql`
```
mysql> SHOW TABLES FROM oqmd_v1_2;
+----------------------------------+
| Tables_in_oqmd_v1_2              |
+----------------------------------+
| accounts                         |
| allocations                      |
| allocations_users                |
| atoms                            |
| auth_group                       |
| auth_group_permissions           |
| auth_permission                  |
| authors                          |
| calculations                     |
| calculations_element_set         |
| calculations_hubbard_set         |
| calculations_meta_data           |
| calculations_potential_set       |
| compositions                     |
| compositions_element_set         |
| compositions_meta_data           |
| django_admin_log                 |
| django_content_type              |
| django_session                   |
| django_site                      |
| dos                              |
| dos_meta_data                    |
| elements                         |
| entries                          |
| entries_element_set              |
| entries_meta_data                |
| entries_project_set              |
| entries_species_set              |
| expt_formation_energies          |
| fits                             |
| fits_dft                         |
| fits_elements                    |
| fits_experiments                 |
| formation_energies               |
| formation_energies_equilibrium   |
| google_analytics_analytics       |
| google_analytics_googleanalytics |
| hosts                            |
| hubbard_corrections              |
| hubbards                         |
| jobs                             |
| journals                         |
| meta_data                        |
| operations                       |
| projects                         |
| projects_allocations             |
| projects_users                   |
| prototypes                       |
| publications                     |
| publications_author_set          |
| reference_energies               |
| rotations                        |
| sites                            |
| spacegroups                      |
| spacegroups_centering_vectors    |
| spacegroups_operations           |
| species                          |
| structures                       |
| structures_element_set           |
| structures_meta_data             |
| structures_species_set           |
| tasks                            |
| tasks_project_set                |
| translations                     |
| users                            |
| users_groups                     |
| users_user_permissions           |
| vasp_potentials                  |
| wyckoffsites                     |
+----------------------------------+
69 rows in set (0.00 sec)

mysql> SELECT COUNT(*) FROM oqmd_v1_2.entries;
+----------+
| COUNT(*) |
+----------+
|   824910 |
+----------+
1 row in set (0.07 sec)
```

(2) Check the OQMD dataset in `qmpy`
```
from qmpy import *

all_entries = Entry.objects.all() \
                   .exclude(duplicate_of=None) \
                   .values_list('duplicate_of', flat=True) \
                   .order_by('duplicate_of') \
                   .distinct()

print(len(all_entries))
```
This script gives the number of entries (687,243).

**Comment by Tony-Y** [comment]
The  password must be placed after `-p` without a space:
```
>> zcat ./qmdb__v1_2__062018.sql.gz | mysql -u dell -pdell_123456 oqmd_v1_2
```

[MySQL Password Option](https://dev.mysql.com/doc/refman/5.7/en/connection-options.html#option_general_password)

**Comment by Tony-Y** [comment]
https://manios.org/2013/03/19/import-a-large-sql-dump-file-to-a-mysql-database-from-command-line

This article might help you.

**Comment by WdBlink** [comment]
hooray！Thanks to your help I find out the problem of my database. The bug comes from my hard disk didn't have enough space. After I cleaned some files, the database reload successfully.
```
mysql> show tables;
+----------------------------------+
| Tables_in_oqmd_v1_2              |
+----------------------------------+
| accounts                         |
| allocations                      |
| allocations_users                |
| atoms                            |
| auth_group                       |
| auth_group_permissions           |
| auth_permission                  |
| authors                          |
| calculations                     |
| calculations_element_set         |
| calculations_hubbard_set         |
| calculations_meta_data           |
| calculations_potential_set       |
| compositions                     |
| compositions_element_set         |
| compositions_meta_data           |
| django_admin_log                 |
| django_content_type              |
| django_session                   |
| django_site                      |
| dos                              |
| dos_meta_data                    |
| elements                         |
| entries                          |
| entries_element_set              |
| entries_meta_data                |
| entries_project_set              |
| entries_species_set              |
| expt_formation_energies          |
| fits                             |
| fits_dft                         |
| fits_elements                    |
| fits_experiments                 |
| formation_energies               |
| formation_energies_equilibrium   |
| google_analytics_analytics       |
| google_analytics_googleanalytics |
| hosts                            |
| hubbard_corrections              |
| hubbards                         |
| jobs                             |
| journals                         |
| meta_data                        |
| operations                       |
| projects                         |
| projects_allocations             |
| projects_users                   |
| prototypes                       |
| publications                     |
| publications_author_set          |
| reference_energies               |
| rotations                        |
| sites                            |
| spacegroups                      |
| spacegroups_centering_vectors    |
| spacegroups_operations           |
| species                          |
| structures                       |
| structures_element_set           |
| structures_meta_data             |
| structures_species_set           |
| tasks                            |
| tasks_project_set                |
| translations                     |
| users                            |
| users_groups                     |
| users_user_permissions           |
| vasp_potentials                  |
| wyckoffsites                     |
+----------------------------------+
69 rows in set (0.00 sec)
```
There are a few things to note about the version of python packages. The thing is the packages usually are installed with up to date which aren't support python2.7 when I use `pip install ...`.  So the packages should be installed with like `pip install qmpy==1.2.0` and `pip install monty==1.0.3'

**Comment by WdBlink** [comment]
And I shoud add   ' ‘  on the $user like`zcat ./qmdb__v1_2__062018.sql.gz | mysql -u 'dell' -p dell_123456 oqmd_v1_2`

**Comment by Tony-Y** [comment]
I think 
```
mysql -u 'username'
```
is the same as
```
mysql -u username
```

**Comment by Tony-Y** [comment]
I made three commits related to this issue:

https://github.com/Tony-Y/cgnn/commit/5129c65597c33d78a8a2633b5229e949d0fea77f
Fix the qmpy version to 1.2.0

https://github.com/Tony-Y/cgnn/commit/e028652130b9edb941316c15c1987966e84e1f38
Use a secure method for restoring the database

https://github.com/Tony-Y/cgnn/commit/e028652130b9edb941316c15c1987966e84e1f38
Update the installation instruction of Python packages

I conclude that this issue has been resolved. If you have any problem, reopen this issue.

**Comment by WdBlink** [comment]
Okay, I've seen it. Thanks again for your help.
